src/tests/JIT/Directed/Convert/out_of_range_fp_to_int_conversions.cpp

```diff
@@ -0,0 +1,174 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <limits.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <math.h>
+
+#ifdef _MSC_VER
+#define DLLEXPORT __declspec(dllexport)
+#else
+#define DLLEXPORT __attribute__((visibility("default")))
+#endif
+
+typedef enum {
+    CONVERT_BACKWARD_COMPATIBLE,
+    CONVERT_SENTINEL,
+    CONVERT_SATURATING,
+    CONVERT_NATIVECOMPILERBEHAVIOR,
+    CONVERT_MANAGED_BACKWARD_COMPATIBLE_X86_X64,
+    CONVERT_MANAGED_BACKWARD_COMPATIBLE_ARM32,
+} FPtoIntegerConversionType;
+
+extern "C" DLLEXPORT int32_t ConvertDoubleToLong(double x, FPtoIntegerConversionType t)
```

Comment:
Can we use the name `Int32` rather than `Long`. The name `long` is confusing for .NET developers because its `64-bits`, it's confusing for Unix C/C++ developers because its `32-bits` or `64-bits` (same as `intptr_t`)

Reply:
(Same for other cases, where `UInt32` or other might help provide clarity)

Reply:
:) I converted all this code to have better names.